doc/tools/coqrst/coqdoc/main.py

```diff
@@ -35,7 +35,7 @@
 
 def coqdoc(coq_code, coqdoc_bin=None):
     """Get the output of coqdoc on coq_code."""
-    coqdoc_bin = coqdoc_bin or os.path.join(os.getenv("COQBIN"), "coqdoc")
+    coqdoc_bin = coqdoc_bin or os.path.join(os.getenv("COQBIN") or "", "coqdoc")
     fd, filename = mkstemp(prefix="coqdoc-", suffix=".v")
     if platform.system().startswith("CYGWIN"):
         # coqdoc currently doesn't accept cygwin style paths in the form "/cygdrive/c/..."
```

Comment:
You can pass a default to getenv instead of using `or ""`. But I'm not sure this is a good idea: do you really want to run the system-global coqdoc here?
This code was written with the assumption that callers would either set COQBIN or pass and explicit coqdoc_bin

Reply:
> You can pass a default to getenv instead of using `or ""`. But I'm not sure this is a good idea: do you really want to run the system-global coqdoc here?

Yes, we want to run with the `coqdoc` in `PATH` here; using `COQBIN` has a problem as it doesn't properly compose when building several packages together.

Should I update and remove the `or` then?

Reply:
Dune composes builds _à la Nix_ , that is to say, `PATH` is modified so you have the right scope. Starting in Dune 1.6 we can also scope private binaries, which helps with some problems we have with the test suite and `fake_ide`.

Reply:
Yes, changing the or would be good, then. It's too bad that this will make mistakes when running the script on its own trickier to spot, but I don't think that's an important enough issue to delay this PR.

Reply:
I see what you mean, on the other hand I'd dare to say running via path is a pretty common occurrence these days in sandboxed-style builds. For example I think that people on Nix should not have any other `coqtop` in scope.

Having to pass parameters becomes quickly a mess too if you context is large tho, and tends to have portability issues.

Reply:
Amended; I am still open as how to improve the general setup for people that don't like to use `PATH` in general; this is an important issue as we have two core dev setups [Nix and Dune] that rely on it, so we must try to accommodate both options in the best possible way.

Reply:
Thanks. I think the code is OK as is. The problem with picking up the wrong Coq version would only pop up if you ran the script standalone, not as part of the Makefile or through Dune. Do we have to worry about this? I think probably not. If we do want to worry, we could probably add another env var.

Reply:
Indeed, I don't think that we should have to worry about this. Soon Dune will be the only supported way of building BTW.

Reply:
I meant that with this change, sphinx-build alone in the refman directory may yield surprising results. That's not a huge issue.